omglol/paste.go

```diff
@@ -0,0 +1,166 @@
+package omglol
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+)
+
+// Create a Paste object
+func NewPaste(Title, Content string, ModifiedOn int64) *Paste {
+	return &Paste{
+		Title:      Title,
+		Content:    Content,
+		ModifiedOn: ModifiedOn,
+	}
+}
+
+// Returns a string representaion of a Paste
+func (p *Paste) String() string {
+	return fmt.Sprintf("Title: %s, Content: %s, ModifiedOn: %d", p.Title, p.Content, p.ModifiedOn)
+}
+
+// Create a new Paste. See https://api.omg.lol/#token-post-pastes-create-a-new-paste
+func (c *Client) CreatePaste(domain string, paste Paste) error {
+	type pasteRequest struct {
+		Title   string `json:"title"`
+		Content string `json:"content"`
+	}
+
+	p := pasteRequest{
+		Title:   paste.Title,
+		Content: paste.Content,
+	}
+
+	jsonData, err := json.Marshal(p)
+	if err != nil {
+		return err
+	}
+
+	req, err := http.NewRequest(http.MethodPost, fmt.Sprintf("%s/address/%s/pastebin", c.HostURL, domain), bytes.NewBuffer([]byte(jsonData)))
+	if err != nil {
+		return err
+	}
+
+	body, err := c.doRequest(req)
+	if err != nil {
+		return fmt.Errorf("sent: %s, error: %w", jsonData, err)
+	}
+
+	var r apiResponse
+	if err := json.Unmarshal(body, &r); err != nil {
+		fmt.Printf("error unmarshalling response: %v\n", err)
+		return err
+	}
+
+	return nil
+}
+
+// Get a specific paste. See https://api.omg.lol/#noauth-get-pastebin-retrieve-a-specific-paste
+func (c *Client) GetPaste(domain string, pasteTitle string) (*Paste, error) {
+	req, err := http.NewRequest(http.MethodGet, fmt.Sprintf("%s/address/%s/pastebin/%s", c.HostURL, domain, pasteTitle), nil)
+	if err != nil {
+		return nil, err
+	}
+
+	body, err := c.doRequest(req)
+	if err != nil {
+		return nil, err
+	}
+
+	type getpasteResponse struct {
+		Request struct {
+			StatusCode int64 `json:"status_code"`
+			Success    bool  `json:"success"`
+		} `json:"request"`
+		Response struct {
+			Message string `json:"message"`
+			Paste   struct {
+				Title      string `json:"title"`
+				Content    string `json:"content"`
+				ModifiedOn int64  `json:"modified_on"`
+			} `json:"paste"`
+		} `json:"response"`
+	}
+
+	var g getpasteResponse
+	if err := json.Unmarshal(body, &g); err != nil {
+		fmt.Printf("Error unmarshalling response: %v\n", err)
+		return nil, err
+	}
+
+	return NewPaste(g.Response.Paste.Title, g.Response.Paste.Content, g.Response.Paste.ModifiedOn), nil
```

Comment:
Add in `listed` while handling API quirks.
```suggestion
	var listed bool
	if g.Response.Paste.Listed != nil {
		listed = true
	} else {
		listed = false
	}
	
	return NewPaste(g.Response.Paste.Title, g.Response.Paste.Content, listed, g.Response.Paste.ModifiedOn), nil
```